File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/AuthorityPropertyEditor.java
package uk.gov.justice.digital.hmpps.oauth2server.config;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

import java.awt.*;
import java.beans.PropertyChangeListener;
import java.beans.PropertyEditor;


public class AuthorityPropertyEditor implements PropertyEditor {

    private GrantedAuthority grantedAuthority;

    @Override
    public void setValue(final Object value) {
        this.grantedAuthority = (GrantedAuthority) value;
    }

    @Override
    public Object getValue() {
        return grantedAuthority;
    }

    @Override
    public boolean isPaintable() {
        return false;
    }

    @Override
    public void paintValue(final Graphics gfx, final Rectangle box) {

    }

    @Override
    public String getJavaInitializationString() {
        return null;
    }

    @Override
    public String getAsText() {
        return grantedAuthority.getAuthority();
    }

    @Override
    public void setAsText(final String text) throws IllegalArgumentException {
        if (text != null && !text.isEmpty()) {
            this.grantedAuthority = new SimpleGrantedAuthority(text);
        }
    }

    @Override
    public String[] getTags() {
        return new String[0];
    }

    @Override
    public Component getCustomEditor() {
        return null;
    }

    @Override
    public boolean supportsCustomEditor() {
        return false;
    }

    @Override
    public void addPropertyChangeListener(final PropertyChangeListener listener) {

    }

    @Override
    public void removePropertyChangeListener(final PropertyChangeListener listener) {

    }
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AuthorityPropertyEditor.kt
package uk.gov.justice.digital.hmpps.oauth2server.config

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import java.awt.Component
import java.awt.Graphics
import java.awt.Rectangle
import java.beans.PropertyChangeListener
import java.beans.PropertyEditor

class AuthorityPropertyEditor : PropertyEditor {
  private var grantedAuthority: GrantedAuthority? = null

  override fun setValue(value: Any) {
    grantedAuthority = value as GrantedAuthority
  }

  override fun getValue(): Any? = grantedAuthority

  override fun isPaintable(): Boolean = false

  override fun paintValue(gfx: Graphics?, box: Rectangle?) {}
  override fun getJavaInitializationString(): String? = null

  override fun getAsText(): String? = grantedAuthority?.authority

  override fun setAsText(text: String?) {
    if (!text.isNullOrEmpty()) {
      grantedAuthority = SimpleGrantedAuthority(text)
    }
  }

  override fun getTags(): Array<String?> = arrayOfNulls(0)

  override fun getCustomEditor(): Component? = null

  override fun supportsCustomEditor(): Boolean = false

  override fun addPropertyChangeListener(listener: PropertyChangeListener) {}
  override fun removePropertyChangeListener(listener: PropertyChangeListener) {}
}
